### PR TITLE
chore: update link to Pathfinder Framework

### PR DESCRIPTION
**Pull request by raimundo-henriques**
The diff includes the removal of whitespace, but I have "highlighted" the relevant lines with comments.

### CODE REVIEW

File: spec/index.bs [hunk 1]
```diff
@@ -37,14 +37,14 @@ Boilerplate: omit copyright, omit example
...
-    "href": "https://wbcsd.github.io/data-exchange-protocol/v2/",
+    "href": "https://wbcsd.github.io/data-model-extensions/guidance/",
     "title": "Guidance and Criteria Catalog for Pathfinder Data Model Extensions",
     "status": "LS",
     "publisher": "WBCSD"
   },
   "PATHFINDER-FRAMEWORK": {
     "authors": [],
-    "href": "https://www.carbon-transparency.com/media/srhhloun/pathfinder-framework.pdf",
+    "href": "https://wbcsd.github.io/tr/2023/framework-20232601/framework.pdf",
```
Comment (raimundo-henriques) [inline spec/index.bs:47]: Link to Pathfinder Framework

File: spec/index.bs [hunk 2]
```diff
@@ -37,14 +37,14 @@ Boilerplate: omit copyright, omit example
   },
   "EXTENSIONS-GUIDANCE": {
     "authors": [],
-    "href": "https://wbcsd.github.io/data-exchange-protocol/v2/",
+    "href": "https://wbcsd.github.io/data-model-extensions/guidance/",
```
Comment (raimundo-henriques) [inline spec/index.bs:40]: Link to Extensions Guidance

File: guidance/index.bs [hunk 3]
```diff
@@ -84,7 +84,7 @@ Note: Future revisions of the guidance are expected to relax the restriction on
 :: The set of data attributes stemming from [=Pathfinder Framework=] plus its concrete realization as specified in [=Use Case 001 Specification=].
 
 : <dfn>Pathfinder Framework</dfn>
-:: See the Pathfinder [Framework Version 1 Document](https://www.carbon-transparency.com/media/uhtlh3tt/pact-setting-foundations-for-standardized-emissions-data-exchange.pdf).
+:: See the Pathfinder [Framework Version 1 Document](https://wbcsd.github.io/tr/2023/framework-20232601/framework.pdf).
```
Comment (raimundo-henriques) [inline guidance/index.bs:87]: Link to Pathfinder Framework

**Review by bhadley — APPROVED** [review]
Thanks! LGTM!

**Pull request merged**